control-plane/agents/core/src/nexus/specs.rs

```diff
@@ -227,28 +227,34 @@ impl ResourceSpecsLocked {
                     return Err(SvcError::NexusNotFound {
                         nexus_id: request.uuid.to_string(),
                     });
-                } else if spec.share == status.share && spec.share != Protocol::Off {
-                    return Err(SvcError::AlreadyShared {
-                        kind: ResourceKind::Nexus,
-                        id: request.uuid.to_string(),
-                        share: spec.share.to_string(),
-                    });
+                } else {
+                    // validate the operation itself against the spec and status
+                    if spec.share != status.share {
+                        return Err(SvcError::NotReady {
+                            kind: ResourceKind::Nexus,
+                            id: request.uuid.to_string(),
+                        });
+                    } else if request.protocol == spec.share || spec.share.shared() {
```

Comment:
Why do we check that the request protocol matches the spec? Can't we just check for `spec.share.shared()`?

Reply:
Ah yes indeed :+1: